cmd/ocp-build-data-enforcer/main.go

```diff
@@ -0,0 +1,252 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/openshift/imagebuilder"
+	dockercmd "github.com/openshift/imagebuilder/dockerfile/command"
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/github"
+)
+
+type options struct {
+	ocpBuildDataRepoDir string
+}
+
+func gatherOptions() *options {
+	o := &options{}
+	flag.StringVar(&o.ocpBuildDataRepoDir, "ocp-build-data-repo-dir", "../ocp-build-data", "The directory in which the ocp-build-data reposity is")
+	flag.Parse()
+	return o
+}
+func main() {
+	opts := gatherOptions()
+
+	configs, err := gatherAllOCPImageConfigs(opts.ocpBuildDataRepoDir)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to gather all ocp image configs")
+	}
+
+	streamMap, err := readStreamMap(opts.ocpBuildDataRepoDir)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to read streamMap")
+	}
+
+	for cfgIdx := range configs {
+		dereferenceStreams(&configs[cfgIdx], streamMap)
+	}
+
+	errGroup := &errgroup.Group{}
+	var counter int
+	for _, config := range configs {
+		counter++
+		config := config
+		errGroup.Go(func() error {
+			processDockerfile(config)
+			return nil
+		})
+	}
+	if err := errGroup.Wait(); err != nil {
+		logrus.WithError(err).Fatal("Processing failed")
+	}
+	logrus.Infof("Processed %d configs", counter)
+}
+
+func dereferenceStreams(config *ocpImageConfig, streamMap streamMap) {
+	if replacement, hasReplacement := streamMap[config.From.Stream]; hasReplacement && replacement.UpstreamImage != nil {
+		config.From.Stream = *replacement.UpstreamImage
+	}
+	for blder := range config.From.Builder {
+		if replacement, hasReplacement := streamMap[config.From.Builder[blder].Stream]; hasReplacement && replacement.UpstreamImage != nil {
+			config.From.Builder[blder].Stream = *replacement.UpstreamImage
+		}
+	}
+}
+
+func processDockerfile(config ocpImageConfig) {
+	orgRepo := config.orgRepo()
+	log := logrus.WithField("file", config.SourceFileName).WithField("org/repo", orgRepo)
+	split := strings.Split(orgRepo, "/")
+	if n := len(split); n != 2 {
+		log.Errorf("splitting orgRepo didn't yield 2 but %d results", n)
+		return
+	}
+	if split[0] == "openshift-priv" {
+		log.Warn("Ignoring repo in openshift-priv org")
+		return
+	}
+	org, repo := split[0], split[1]
```

Comment:
Are we validate this?

Reply:
Not sure what you mean? A couple of lines above we do a len check and error out, yeah.